src/widget/rating/rating.js

```diff
@@ -0,0 +1,117 @@
+import RangeWidget from '../../widget/range/range-widget';
+import { isNumber } from '../../js/utils';
+
+/**
+ * @extends RangeWidget
+ */
+class RatingWidget extends RangeWidget {
+    /**
+     * @type string
+     */
+    static get selector() {
+        return '.or-appearance-rating input[type="number"]';
+    }
+
+    _init() {
+        const fragment = document.createRange().createContextualFragment( this._getHtmlStr() );
+        this.element.after( fragment );
+        this.element.classList.add( 'hide' );
+        this.widget = this.question.querySelector( '.widget' );
+
+        let ticks = this.props.ticks ? Math.ceil( Math.abs( ( this.props.max - this.props.min ) / this.props.step ) ) : 1;
+        // Now reduce to a number < 50 to avoid showing a sold black tick line.
+        let divisor = Math.ceil( ticks / this.props.maxTicks );
+        while ( ticks % divisor && divisor < ticks ) {
+            divisor++;
+        }
+        ticks = ticks / divisor;
+
+        let addTicks = new Array( ticks ).fill( '<span></span>' ).join( '' );
+        let ratingStars = '',
+            name = Math.random().toString( 36 ).substring( 2, 15 );
+        for ( let i = this.props.min; i <= this.props.max; i += this.props.step ) {
+            ratingStars += `<input type=radio name="rating-stars__${name}" class="star rating__tick" value="${i}"/>`;
+        }
+        addTicks = ratingStars;
+
+        this.widget.querySelector( '.range-widget__ticks' )
+            .append( document.createRange().createContextualFragment( addTicks ) );
+
+        this.widgetThick = this.widget.querySelector( '.range-widget__ticks' );
+        this.widgetThick.querySelectorAll( '.star.rating__tick' ).forEach( el => {
+            el.addEventListener( 'click', () => {
```

Comment:
Since these are `input` elements, I think we should use `change` or `input`. This `click` listener does seem to work, but i have a suspicion we may be missing some special case/device/browser issue.

Reply:
code updated